caikit/core/data_model/dataobject.py

```diff
@@ -274,7 +289,7 @@ def make_dataobject(
             class derived from DataObjectBase with the given name and
             annotations
     """
-    bases = tuple([DataObjectBase] + list(bases or []))
+    bases = tuple([Type[DataObjectBase]] + list(bases or []))
```

Comment:
This is interesting. These `bases` are actually the concrete bases passed to the metaclass' `__new__` function, so I actually _explicitly_ want `DataObjectBase` here. I'm honestly not sure what happens if you pass `Type[BaseClassIWant]` to the `__new__` of a `type`

Reply:
Yeah, I think I misread the code there, due to rather confusing error message.  The error I am trying to fix is:
<img width="996" alt="image" src="https://github.com/caikit/caikit/assets/6641871/ed0063d1-db56-4993-a0fe-20c7304160a4">

After a little more digging, this seems to be some oddity of the tuple construction, it wants a list that is a single type, some sort of co-constraint thing I guess.
<img width="797" alt="image" src="https://github.com/caikit/caikit/assets/6641871/8f47a633-99a8-41fe-862c-c590de477ac8">

Seems to be a type error and not an indication of actual error, though maybe it inhibits an optimization, not sure.

I tried a bunch of things.  This seems to work type-wise and , and is fairly clean and elegant, and doesn't construct intermediary lists:
```python
bases = (DataObjectBase,) + tuple(bases or ())
```

I spent way too much time on this.  :-)

Reply:
ha! Love me a rabbit hole. This looks good, thanks for digging so deep!